src/preact/utils/docInfo.js

```diff
@@ -0,0 +1,46 @@
+import {urlUtils} from './url';
+
+export const docInfo = {
```

Comment:
I believe we'll need to scope this per document, so that we can support cross-frame documents. We can support that with `Context` APIs.

Reply:
(Posted question to slack for clarification)

Reply:
Created https://github.com/ampproject/amphtml/issues/37465 to track this improvement.